### PR TITLE
Fast boot platform plugin

### PR DESCRIPTION
**Pull request by sridhar-ravindran**
**- What I did**
Added support for calling a platform specific plugin in fastboot, warmboot
**- How I did it**
On branch FastBoot_Platform_Plugin
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)
        modified:   scripts/fast-reboot

**- How to verify it**
Write a platform plugin script and execute fast-reboot command
Plugin script will be executed and then reboot will be invoked
**- Previous command output (if the output of a command-line utility has changed)**

**- New command output (if the output of a command-line utility has changed)**

-->

### CODE REVIEW

File: scripts/fast-reboot [hunk 1]
```diff
@@ -13,6 +13,9 @@ STRICT=no
 REBOOT_METHOD="/sbin/kexec -e"
 ASSISTANT_IP_LIST=""
 ASSISTANT_SCRIPT="/usr/bin/neighbor_advertiser"
+DEVPATH="/usr/share/sonic/device"
+PLATFORM=$(sonic-cfggen -H -v DEVICE_METADATA.localhost.platform)
+PLATFORM_PLUGIN="platform_plugin"
```
Comment (jleveque) [inline scripts/fast-reboot]: The name "platform_plugin" is vague. Suggest to change to "fastboot_plugin" or similar?

Comment (santhosh-kt) [reply scripts/fast-reboot]: Changed to "fastboot_plugin". Please approve the changes.

**Comment by jleveque** [comment]
Retest this please

File: scripts/fast-reboot [hunk 2]
```diff
@@ -502,6 +505,11 @@ if [ -x /sbin/hwclock ]; then
     /sbin/hwclock -w || /bin/true
 fi
 
+if [ -x ${DEVPATH}/${PLATFORM}/${PLATFORM_PLUGIN} ]; then
+    VERBOSE=yes debug "Running ${PLATFORM} specific plugin..."
```
Comment (yxieca) [inline scripts/fast-reboot]: Can you remove "VERBOSE=yes"? We would like to keep console quiet in normal execution.

**Comment by yxieca** [comment]
What is planned use case for this plugin?

File: scripts/fast-reboot [hunk 3]
```diff
@@ -13,6 +13,9 @@ STRICT=no
 REBOOT_METHOD="/sbin/kexec -e"
 ASSISTANT_IP_LIST=""
 ASSISTANT_SCRIPT="/usr/bin/neighbor_advertiser"
+DEVPATH="/usr/share/sonic/device"
+PLATFORM=$(sonic-cfggen -H -v DEVICE_METADATA.localhost.platform)
+PLATFORM_PLUGIN="fastboot_plugin"
```
Comment (yxieca) [inline scripts/fast-reboot]: You can change the PLUGIN name to "${REBOOT_TYPE}_plugin" so that warm reboot and fast reboot could have different plugins.

Comment (lguohan) [reply scripts/fast-reboot]: what is the reason to have fastboot_plugin per platform?

**Comment by sridhar-ravindran** [comment]
Hi,
We have raised https://github.com/Azure/sonic-utilities/pull/740 with updated code changes.
Request you to review the same. Closing this PR

**Pull request closed without merging**